src/Cli/LegacyParamArgvInput.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+namespace SilverStripe\Cli;
+
+use SilverStripe\Dev\Deprecation;
+use SilverStripe\ORM\ArrayLib;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\InputDefinition;
+
+/**
+ * Represents an input coming from the CLI arguments - but converts legacy arg-style parameters to flags.
+ *
+ * e.g. `ddev dev:build flush=1` is converted to `ddev dev:build --flush`.
+ * Doesn't convert anything that isn't explicitly an InputOption in the relevant InputDefinition.
+ * Removes the parameters from the input args (e.g. doesn't become `ddev dev:build flush=1 --flush`).
+ *
+ * @deprecated 6.0.0 Use Symfony\Component\Console\Input\ArgvInput instead.
+ */
+class LegacyParamArgvInput extends ArgvInput
+{
+    /**
+     * Input from the command line.
+     *
+     * We need a separate copy of this because the one held by the parent class is private
+     * and not exposed until symfony/console 7.1
+     */
+    private array $argv;
+
+    public function __construct(?array $argv = null, ?InputDefinition $definition = null)
+    {
+        Deprecation::withNoReplacement(
```

Comment:
That stuff hasn't been merged up yet.

Reply:
Should rebase it in then so we don't have things break later

Reply:
It hadn't been merged up from 5 to 6. I can't rebase on top of something that isn't in the branch I need to rebase on??
Since those comments were made it _has_ been merged up, so _now_ I'm able to make the change. Though it really shouldn't have been blocked for merging for something that wasn't even in the branch I'm targetting 😅

Reply:
Done